Library/Homebrew/formula_creator.rb

```diff
@@ -82,7 +82,10 @@ def write_formula!
           r = Resource.new
           r.url(@url)
           r.owner = self
-          @sha256 = r.fetch.sha256 if r.download_strategy == CurlDownloadStrategy
```

Comment:
This feels like a change in functionality?

Reply:
The whole block is `unless @head`, and if `--HEAD` is not given, the URL is supposed to be a file.
```
      --HEAD                       Indicate that URL points to the package's
                                   repository rather than a file.
```

Reply:
Ok thanks